pypowsybl/network.py

```diff
@@ -186,6 +186,12 @@ def __init__(self, handle: _pp.JavaHandle):
         self._source_format = att.source_format
         self._forecast_distance = _datetime.timedelta(minutes=att.forecast_distance)
         self._case_date = _datetime.datetime.utcfromtimestamp(att.case_date)
+        self.elements = {"Bus":self.get_buses().shape[0],"Busbarsections":self.get_busbar_sections().shape[0],
```

Comment:
That information will become wrong as soon as the user changes something in the network:
it's not the right place to do it

Reply:
@sylvlecl I am trying to figure out the right place to do it , can you help me pls

Reply:
I think we could retrieve the information where we actually use it, which would mean in __str__.

That's why it needs to be a pretty "light" operation, we don't want a call to `str(network)` to take long.
For this to be efficient, it must not retrieve the whole dataframes (this implies some quite heavy data transfer between the java, object, representation and the C structure underlying the dataframe), but it should instead only get the size information from the java network. So this means adding methods in the C layer and java layer (NetworkCFunctions.java) for this, for example to get the count of elements for a given element type.